### PR TITLE
Feature rpc server

### PR DESCRIPTION
**Pull request by antoinealb**
I need some testing but this is the initial implementation of a service call server for the master board. If you guys can review it a bit it would be nice.

### CODE REVIEW

File: src/rpc_server.c [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+#define RPC_SERVER_STACKSIZE 2048
+#define RPC_SERVER_PORT 20001
+THD_WORKING_AREA(wa_rpc_server, RPC_SERVER_STACKSIZE);
+
+uint8_t input_buffer[1024];
+uint8_t output_buffer[1024];
+
+bool method_called;
+size_t output_bytes_written;
```
Comment (Stapelzeiger) [inline src/rpc_server.c]: these variables should be static

File: src/rpc_server.c [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        method_called = false;
+        output_bytes_written = 0;
+
+        while (!method_called) {
+            /* Tries to receive something from the connection. */
+            err = netconn_recv(client_conn, &buf);
+
+            /* If connection closed early, exit the thread. */
+            if (err != ERR_OK) {
+                break;
```
Comment (Stapelzeiger) [inline src/rpc_server.c]: in this case the netconn_close & delete are not called

File: src/rpc_server.c [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+        /* method_called will be set to true once a callback is fired. */
+        method_called = false;
+        output_bytes_written = 0;
+
+        while (!method_called) {
+            /* Tries to receive something from the connection. */
+            err = netconn_recv(client_conn, &buf);
+
+            /* If connection closed early, exit the thread. */
```
Comment (Stapelzeiger) [inline src/rpc_server.c]: what thread?

**Comment by antoinealb** [comment]
Fixed the remarks of @Stapelzeiger thanks for the review

**Pull request merged**